### ISSUE TITLE
enclave to "secure active delegation"

### ISSUE THREAD

**Issue by aeyakovenko** [solana-labs/solana#2149]
#### Problem

Enclave is very specific to an implementation feature.

#### Proposed Solution

"Secure Active Delegation", is a design to reduce the attack vectors that could compromise a key that is used for fork selection voting.

Future work:
1. Allow for staked clients to connect to untrusted validators and actively vote on validation messages.
* reduce dependencies in the code.  Goal should be #[no_std], no allocation. We want to ensure that this part of the code base is fully audited.

2.  Exploits that cause untrusted code execution in the validator.  Validators should be designed such that each execution environment is as different as possible to reduce the likely-hood of an exploit producing consistent state.  If the exploit results in inconsistent state, then active delegator will observe it and fail to vote.  (tx and blob execution randomization, any place we can do things in parallel, should be done in random order)

3. Monitoring and options for maintaining "decentralization".  We don't want everyone in the world to actively delegate to the same validator.  So active delegation should potentially switch and split between validators.
 

tag: @pgarg66 

Feel free to propose more ideas.
